python/dgl/heterograph.py

```diff
@@ -6065,6 +6065,88 @@ def local_scope(self):
+            the cloned graph's sparse formats will remain consistent with the
+            original.
+          * If a format in the original graph is also in ``allowed_formats``,
+            any formats not in ``allowed_formats`` will be removed in the cloned
+            graph.
+          * If the original graph has no formats in ``allowed_formats``, a new
+            format will be made for the cloned graph. This will be in the order
+            of ``'coo'``, ``'csr'``, ``'csc'``, regardless of the
+            ``allowed_formats`` order.
+
+        Parameters
+        ----------
+        allowed_formats : str or list of str or None
+
+            * If ``allowed_formats`` is None, return the usage status of sparse
+              formats.
+            * Otherwise, it can be a string ``'coo'``, ``'csr'``, or ``'csc'``
+              or a sublist of them, specifying the allowed sparse formats to
+              use.
+
+        create_formats : str or list of str or None
+
+            * If ``create_formats`` is None:
+              * If ``allowed_formats`` is None, return the usage status of
+                sparse formats.
+              * Otherwise, follow the rules above to create sparse matrices for
+                the cloned graph.
+            * Otherwise, it can be ``'coo'``, ``'csr'``, or ``'csc'`` or a
+              sublist of them, specifying the sparse matrices to create. If
+              ``allowed_formats`` is None or there are some formats in the list
+              that are not allowed for the graph, an error will be raised.
+
+        Returns
+        -------
+        dict or DGLGraph
+
+            * If ``allowed_formats`` is None, the result will be a dict
+              recording the usage status of sparse formats.
+            * Otherwise, a DGLGraph will be returned, which is a clone of the
+              original graph with the specified allowed sparse format(s) and
+              created sparse format(s).
+        """
+        pass
+
+    def new_create_formats_(self, create_formats=None):
+        r"""Create specified sparse matrices allowed for the graph.
+
+        By default, sparse matrices for a graph are created only when necessary.
+        In some cases, such as in a multi-process data loader, we may want to
+        create them immediately. This API allows for that.
+
+        Parameters
+        ----------
+        create_formats : str or list of str or None
+
+            * If ``create_formats`` is None, create all sparse matrices allowed
+              for the graph.
+            * Otherwise, it can be ``'coo'``, ``'csr'``, or ``'csc'`` or a
+              sublist of them, specifying the sparse matrices to create. If
+              there are some formats already created, they will be skipped.
+              If there are some formats in the list that are not allowed for
+              the graph, an error will be raised.
+        """
+        pass
```

Comment:
What are the advantages of this API over first specifying the allowed formats with one API and then create all allowed formats with another API?

Reply:
the formats() method are designed for out-place reset, and create_formats_ is designed for in-place creation, that's why we have 2 different apis

Reply:
Still, that's already what `g.create_formats_()` was doing right? The only differences are that this new API allows creating for a subset of allowed formats that's not necessarily a subset of the allowed formats, which might raise trigger additional errors. The benefits seem to be quite small in my opinion. Why not stick to `g.create_formats_()` for now and add a new API if this becomes really necessary?

Reply:
Indeed, but I'm considering that under the condition of `create_formats = None`, the original behavior of `g.create_formats_()` will not be disrupted. 

For those who wish to modify in place, `create_formats` might be quite helpful. As for accidental errors, such as when some format(s) in `create_formats` are not in the graph's `allowed_formats`, I believe these types of exceptions are easy to catch and return with a clear error message.

Reply:
Yes. I just want to point out that when there is not a clear customer, it may not be worth the efforts to expand the current API.

Reply:
Oh, sorry for the confusion, this PR is just for code review on the API doc string. Eventually, the old formats and create_formats_ will be replace by these 2 apis.

We will never name a API newblabla, new will be old right after it get merged :)

Reply:
I understand. The point is that you may not even need to modify that API at all.